robot/impl/resource_manager.go

```diff
@@ -529,6 +530,153 @@ func (manager *resourceManager) completeConfig(
 	}()
 
 	// first handle remotes since they may reveal unresolved dependencies
+	manager.completeConfigForRemotes(ctx, lr)
+
+	// now resolve prior to sorting in case there's anything newly discovered
+	if err := manager.resources.ResolveDependencies(manager.logger); err != nil {
+		// debug here since the resolver will log on its own
+		manager.logger.CDebugw(ctx, "error resolving dependencies", "error", err)
+	}
+
+	levels := manager.resources.ReverseTopologicalSortInLevels()
+	timeout := rutils.GetResourceConfigurationTimeout(manager.logger)
+	for _, resourceNames := range levels {
+		// Q: can we use `reconfigureWorkers` instead?
```

Comment:
we should only be running one instance of this routine concurrently at a time for a given local robot, so I think this we can use this existing `WaitGroup`, which is only waited on in tests at the moment